### PR TITLE
Fix error in Thirdweb Auth > Firebase code

### PR DESCRIPTION
**Pull request by kien-ngo**
The code in this particular section of the docs doesn't work, simply because `usersRef` will always be an object, which mean `if (!usersRef) {}` will never be accessed (since it's falsy).
Link: https://portal.thirdweb.com/auth/integrations/firebase#sign-up--log-in-users
image

There are other proper ways to check if a document exists in Firestore, including `snapshot.exists()` (check the image below)
image

### CODE REVIEW

**Comment by adam-maj** [comment]
Amazing @kien-ngo, thanks a lot for identifying this issue! Huge help, much appreciated. If you'd like to, you could lead the charge on updating the actual example project code as well and put up a PR there: https://github.com/thirdweb-example/firebase-auth

Otherwise I'll get to it later today/tm.

**Comment by kien-ngo** [comment]
I can handle it @adam-maj ^

**Comment by kien-ngo** [comment]
@adam-maj Actually the code is the thirdweb example is already correct. It also has cleaner syntax than my PR's, so I will sync that code from the example to my PR
image

File: docs/onboarding/10 Auth/6 Integrations/2 Firebase.mdx [hunk 1]
```diff
@@ -150,36 +150,37 @@ The `signIn` function:
...
-      // If this is a new user, we create a new document in the database.
-      const usersRef = doc(db, "users", user.uid!);
-      if (!usersRef) {
-        // User now has permission to update their own document outlined in the Firestore rules.
-        setDoc(usersRef, { createdAt: serverTimestamp() }, { merge: true });
-      }
-    })
-    .catch((error) => {
-      console.error(error);
+  const payload = await thirdwebAuth?.login();
```
Comment (kien-ngo) [inline docs/onboarding/10 Auth/6 Integrations/2 Firebase.mdx:153]: The code below is copied directly from the code in https://github.com/thirdweb-example/firebase-auth as suggested by Adam

**Comment by warengonzaga** [comment]
Can you take a look into this @MananTank? Thanks!

**Comment by kien-ngo** [comment]
cc @saminacodes @ciaranightingale

**Pull request merged**